src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -22,7 +29,12 @@ export type TopNavMenuProps<QT extends Query | AggregateQuery = Query> =
   StatefulSearchBarProps<QT> &
     Omit<SearchBarProps<QT>, 'kibana' | 'intl' | 'timeHistory'> & {
       config?: TopNavMenuData[];
-      badges?: Array<EuiBadgeProps & { badgeText: string }>;
+      badges?: Array<
+        EuiBadgeProps & {
+          badgeText: string;
+          toolTipProps: EuiToolTipProps;
```

Comment:
**nit:** Might be nicer if this was of type `Partial<EuiTooltipProps>` - that way, we could remove the `as EuiToolTipProps` in [dashboard_top_nav.tsx](https://github.com/elastic/kibana/pull/154253/files#diff-bdff15664bb834abd58d9a88f105bc6fa74513686726f5fc5110720201ee6a1dR260)